lib/ueberauth/strategy/facebook.ex

```diff
@@ -16,16 +22,20 @@ defmodule Ueberauth.Strategy.Facebook do
   Handles initial request for Facebook authentication.
   """
   def handle_request!(conn) do
-    auth_type = option(conn.params["auth_type"], conn, :auth_type)
-    scopes = option(conn.params["scope"], conn, :default_scope)
+    params = conn.params
```

Comment:
You could reduce the number of matches if you refactor this into a pipechain.  Not idea if this is better or worse:

```elixir
allowed_params = conn
                 |> option(:allowed_request_params)
                 |> Enum.map(&to_string/1)

opts = conn.params
       |> maybe_replace_param(conn, "auth_type", :auth_type)
       |> maybe_replace_param(conn, "scope", :default_scope)
       |> Enum.filter(fn {k,_v} -> Enum.member?(allowed_params, k) end)
       |> Enum.map(fn {k,v} -> {String.to_existing_atom(k), v} end)
       |> Keyword.put(:redirect_uri, callback_url(conn))
```